core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/cluster/rds/upgrade/UpgradeRdsService.java

```diff
@@ -110,11 +110,12 @@ public void handleInstallPostgresPackagesError(Long stackId, String version, Str
     public void rdsUpgradeFinished(Long stackId, Long clusterId, TargetMajorVersion targetMajorVersion) {
         String statusReason = "RDS upgrade finished";
         LOGGER.debug(statusReason);
+        stackUpdater.updateExternalDatabaseEngineVersion(stackId, targetMajorVersion.getMajorVersion());
+        rdsUpgradeOrchestratorService.updateDatabaseEngineVersion(stackId, targetMajorVersion.getMajorVersion());
```

Comment:
it's calling a long running salt related stuff from an action instead of a handler. we usually do this via handler. could you revisit if this could be moved to a handler?